src/public/services/UserService.ts

```diff
@@ -12,3 +16,22 @@ export const STORAGE_USER_KEY = 'user'
 export const saveUserToLocalStorage = (user: User | null): void => {
   localStorage.setItem(STORAGE_USER_KEY, JSON.stringify(user))
 }
+
+/**
+ * Fetches the user from the server using the current session cookie.
+ *
+ * Side effect: Saves the retrieved user to localStorage
+ * @returns the logged in user if session is valid, `null` otherwise
+ */
+export const fetchUser = async (): Promise<User | null> => {
+  return axios
+    .get<User>(USER_ENDPOINT)
+    .then(({ data: user }) => {
+      saveUserToLocalStorage(user)
```

Comment:
you can move this into the controller if you decide to do so in the previous PR

Reply:
donezxc